### PR TITLE
feat(erc1155): harden script to only mint to recipients with balance 0

### PR DESCRIPTION
**Pull request by dohaki**
Adds check to mint script to skip addresses that already have received a token

### CODE REVIEW

File: scripts/mintERC1155.ts [hunk 1]
```diff
@@ -27,18 +27,31 @@ async function main() {
...
+      console.log(`Skipping ${alreadyMintedRecipients.length} already minted recipients...`);
+    }
+
+    if (recipientsToMint.length === 0) {
+      console.log(`No recipients to mint for. Skipping chunk...`);
+      continue;
+    }
+
+    const airdropTx = await erc1155.airdrop(tokenId, recipientsToMint, 1);
+    console.log(`Minting token with id ${tokenId} to ${recipientsToMint.length} recipients...`);
```
Comment (james-a-morris) [inline scripts/mintERC1155.ts:51]: Nit: should we place this before we run the airdrop tx? If it fails to perform the airdrop we would have this as the last log

**Review by james-a-morris — APPROVED** [review]
Left a single nit. this is great code

File: scripts/mintERC1155.ts (hunk 1 above)
Comment (dohaki) [reply scripts/mintERC1155.ts:50]: Good catch. Changed

File: scripts/mintERC1155.ts [hunk 2]
```diff
@@ -27,18 +27,31 @@ async function main() {
...
+    if (recipientsToMint.length === 0) {
+      console.log(`No recipients to mint for. Skipping chunk...`);
+      continue;
+    }
+
+    console.log(`Minting token with id ${tokenId} to ${recipientsToMint.length} recipients...`);
+    const airdropTx = await erc1155.airdrop(tokenId, recipientsToMint, 1);
+    console.log("Tx hash:", airdropTx.hash);
+    await airdropTx.wait();
+    console.log(`Successfully minted token to ${recipientsToMint.length} recipients in chunk`);
```
Comment (pxrl) [inline scripts/mintERC1155.ts:54]: Consider dropping the txn hash here?

**Pull request merged**